src/authentication/create-account-method/container.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+import { Action } from 'redux';
+import { RootState } from '../../store/reducer';
+import { connectContainer } from '../../store/redux-container';
+import { RegistrationStage, registerWithEmail, registerWithWallet } from '../../store/registration';
+
+import { CreateAccountMethod } from '.';
+
+export interface CreateAccountMethodProps {
+  stage: RegistrationStage;
+  isConnecting: boolean;
+
+  registerWithEmail: () => Action<string>;
+  registerWithWallet: () => Action<string>;
+}
+
+export class Container extends React.Component<CreateAccountMethodProps> {
+  static mapState(state: RootState): Partial<CreateAccountMethodProps> {
+    const { registration } = state;
+    return {
+      stage: registration.stage,
+      isConnecting: registration.loading,
+    };
+  }
+
+  static mapActions(_props: CreateAccountMethodProps): Partial<CreateAccountMethodProps> {
+    return { registerWithEmail: registerWithEmail, registerWithWallet: registerWithWallet };
+  }
+
+  handleSelectionChange = (selectedOption: string) => {
+    this.setState({ selectedOption });
```

Comment:
:thinking: I'm surprised there isn't a typescript error here as we didn't define the state?

Actually... do we use it?

Reply:
good catch thanks, it was not being used!